cmd/kube-scheduler/app/testing/command_test.go

```diff
@@ -0,0 +1,361 @@
+					SchedulerName: v1.DefaultSchedulerName,
+					Plugins:       defaultPlugins,
+				},
+			},
+		},
+		{
+			name: "v1alpha1 config with SchedulerName and HardPodAffinitySymmetricWeight",
+			flags: []string{
+				"--config", v1alpha1Config,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: schedulerName,
+					Plugins:       defaultPlugins,
+				},
+			},
+		},
+		{
+			name: "plugin config with single profile",
+			flags: []string{
+				"--config", pluginConfigFile,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: v1.DefaultSchedulerName,
+					Plugins: map[string][]kubeschedulerconfig.Plugin{
+						"BindPlugin":      {{Name: "DefaultBinder"}},
+						"FilterPlugin":    {{Name: "NodeResourcesFit"}, {Name: "NodePorts"}},
+						"PreFilterPlugin": {{Name: "NodeResourcesFit"}, {Name: "NodePorts"}},
+						"PreScorePlugin":  {{Name: "InterPodAffinity"}, {Name: "TaintToleration"}},
+						"QueueSortPlugin": {{Name: "PrioritySort"}},
+						"ScorePlugin":     {{Name: "InterPodAffinity", Weight: 1}, {Name: "TaintToleration", Weight: 1}},
+					},
+				},
+			},
+		},
+		{
+			name: "plugin config with multiple profiles",
+			flags: []string{
+				"--config", multiProfilesConfig,
+				"--kubeconfig", configKubeconfig,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: profileName1,
+					Plugins:       defaultPlugins,
+				},
+				{
+					SchedulerName: profileName2,
+					Plugins: map[string][]kubeschedulerconfig.Plugin{
+						"BindPlugin":      {{Name: "DefaultBinder"}},
+						"QueueSortPlugin": {{Name: "PrioritySort"}},
+					},
+				},
+			},
+		},
+		{
+			name: "Deprecated SchedulerName flag",
+			flags: []string{
+				"--kubeconfig", configKubeconfig,
+				"--scheduler-name", schedulerName,
+			},
+			wantProfiles: []schedulerProfile{
+				{
+					SchedulerName: schedulerName,
+					Plugins:       defaultPlugins,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Clean up config for next test case
+			defer configz.Delete("componentconfig")
+
+			fs := pflag.NewFlagSet("test", pflag.PanicOnError)
+			opts, err := options.NewOptions()
+			for _, f := range opts.Flags().FlagSets {
+				fs.AddFlagSet(f)
+			}
+			if err := fs.Parse(tc.flags); err != nil {
+				t.Fatal(err)
+			}
+			cmd := &cobra.Command{}
+			args := []string{}
+			cc, err := app.SetupCommand(cmd, args, opts)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer cc.SecureServing.Listener.Close()
+			defer cc.InsecureServing.Listener.Close()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			recorderFactory := profile.NewRecorderFactory(events.NewBroadcaster(&events.EventSinkImpl{Interface: cc.EventClient.Events("")}))
+			outOfTreeRegistry := make(framework.Registry)
+			// Create the scheduler.
+			sched, err := scheduler.New(cc.Client,
```

Comment:
This shouldn't be calling `scheduler.New`. Instead, the part of `app.Run` that calls `scheduler.New` should be part of `setupCommand`. And `Run` should do exactly that: call `sched.Run`.